datafusion/expr/src/utils.rs

```diff
@@ -901,7 +901,7 @@ pub fn can_hash(data_type: &DataType) -> bool {
         DataType::UInt64 => true,
         DataType::Float32 => true,
         DataType::Float64 => true,
-        DataType::Timestamp(time_unit, None) => match time_unit {
+        DataType::Timestamp(time_unit, _) => match time_unit {
```

Comment:
it seems like there is no reason to check the timeunit either (as all branches are true)

However that is not something introduced in this PR